app/services/TwoStepVerification.scala

```diff
@@ -55,14 +62,22 @@ trait TwoStepVerification {
           }
         }
         shouldRedirectTo2SV.map {
-          case true => Some(wrapLocationWith2SV(continue))
+          case true =>
+            twoStepVerificationThrottle.registrationMandatory(authContext.user.oid) match {
+              case true =>
+                auditContext.setSentToMandatory2SVRegister()
+                Some(wrapLocationWith2SV(continue, twoStepVerificationRequiredUrl))
+              case _ =>
+                auditContext.setSentToOptional2SVRegister()
+                Some(wrapLocationWith2SV(continue, continue.fullUrl))
+            }
           case _ => None
         }
-      }.andThen { case Success(Some(_)) => auditContext.sentTo2SVRegister = true }
+      }
     } else Future.successful(None)
   }
 
-  private def wrapLocationWith2SV(continue: Location) = Locations.twoStepVerification(Map("continue" -> continue.fullUrl, "failure" -> continue.fullUrl) ++
+  private def wrapLocationWith2SV(continue: Location, failure: String) = Locations.twoStepVerification(Map("continue" -> continue.fullUrl, "failure" -> failure) ++
```

Comment:
I understand this is a private method but can both the parameters be of the same type?